### PR TITLE
Allow query comments and comment counts by tags

### PR DESCRIPTION
**Pull request by alinhle**
## What does this PR do?
- Allow querying asset comments and comment counts by tags.

### CODE REVIEW

File: graph/resolvers/asset.js [hunk 1]
```diff
@@ -4,31 +4,32 @@ const Asset = {
...
       parent_id: null,
       excludeIgnored,
     });
   },
-  commentCount({id, commentCount}, {excludeIgnored}, {user, loaders: {Comments}}) {
-    
+  commentCount({id, commentCount}, {query: {excludeIgnored, tag}}, {user, loaders: {Comments}}) {
+
     // TODO: remove
     if (user && excludeIgnored) {
```
Comment (alinhle) [inline graph/resolvers/asset.js]: This if clause is not right yet :-P

File: graph/resolvers/asset.js [hunk 2]
```diff
@@ -4,31 +4,32 @@ const Asset = {
...
     if (commentCount != null) {
       return commentCount;
     }
     return Comments.parentCountByAssetID.load(id);
   },
-  totalCommentCount({id, totalCommentCount}, {excludeIgnored}, {user, loaders: {Comments}}) {
+  totalCommentCount({id, totalCommentCount}, {query: {excludeIgnored, tag}}, {user, loaders: {Comments}}) {
 
     // TODO: remove
     if (user && excludeIgnored) {
```
Comment (alinhle) [inline graph/resolvers/asset.js]: Same

File: graph/loaders/comments.js [hunk 3]
```diff
@@ -218,6 +218,14 @@ const getCommentCountByQuery = (context, {ids, statuses, asset_id, parent_id, au
     query = query.where({author_id});
   }
 
+  if (tag) {
+    query = query.find({
+      'tags.tag.name': {
+        $in: tag,
```
Comment (wyattjoh) [inline graph/loaders/comments.js]: I think that `tag` should be `tags`. With regards to the performance issues, I think we may need to approach that later in a performance related PR so we can define our indexes.

Comment (alinhle) [reply graph/loaders/comments.js]: Yeah my first thought was `tags` too. Someone already added that to the definitions (without implementing the loader) to the graphql and used `tag` so I thought maybe it had some good reasoning :-)

Comment (wyattjoh) [reply graph/loaders/comments.js]: Sounds good! Feel free to change it up!

**Review by wyattjoh — APPROVED** [review]

**Pull request merged**